src/components/stepforms/WidgetList.vue

```diff
@@ -0,0 +1,165 @@
+<template>
+  <div class="widget-list__container">
+    <label :for="id">{{name}}</label>
+    <div class="widget-list__body">
+      <div class="widget-list__child" v-for="(child, index) in children" :key="index">
+        <div class="widget-list__component">
+          <component :is="widget" :value="child.value" @input="updateChildValue($event, index)"></component>
+        </div>
+        <div class="widget-list__icon" v-if="child.isRemovable" @click="removeChild(index)">
+          <i class="far fa-trash-alt"></i>
+        </div>
+      </div>
+      <button v-if="!automaticNewField" class="widget-list__add-fieldset" @click="addFieldSet">
+        <i class="fas fa-plus-circle"></i>
+        {{ addFieldName }}
+      </button>
+    </div>
+  </div>
+</template>
+<script lang="ts">
+import { Component, Prop, Vue } from 'vue-property-decorator';
+import WidgetInputText from './WidgetInputText.vue';
+import WidgetAggregation from './WidgetAggregation.vue';
+
+type Field = {
+  name: string;
+  label?: string;
+  value: any;
+};
+
+type RepeatableField = Field[];
+
+@Component({
+  name: 'widget-list',
+  components: {
+    WidgetAggregation,
+    WidgetInputText,
+  },
+})
+export default class WidgetList extends Vue {
+  @Prop({ type: String, default: '' })
+  addFieldName!: string;
+
+  @Prop({ type: String, default: null })
+  id!: string;
+
+  @Prop({ type: String, default: '' })
+  name!: string;
+
+  @Prop({ type: String, default: '' })
+  placeholder!: string;
+
+  @Prop({ type: Array, default: () => [] })
+  value!: any[];
+
+  @Prop({ type: String, default: 'widget-input-text' })
+  widget!: string;
+
+  @Prop({ type: Boolean, default: true })
+  automaticNewField!: boolean;
+
+  @Prop({ default: null })
+  defaultItem!: string | RepeatableField;
+
+  get children() {
+    const children: object[] = [];
+    const valueCopy = [...this.value];
+    if (this.automaticNewField) {
+      valueCopy.push(this.defaultChildValue);
+    }
+    valueCopy.forEach(value => {
+      children.push({
+        isRemovable: valueCopy.length !== 1,
+        value,
+      });
+    });
+    return children;
+  }
```

Comment:
What about

```typescript
    return valueCopy.map(value => ({
        isRemovable: valueCopy.length !== 1,
        value,
    });
```

and then removing `const children: object[] = [];` declaration?